src/components/rules-creation-page/steps/IncludeExcludeSelectionStep.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useState, useEffect } from "react";
+import { withSnackbar } from "d2-ui-components";
+import SyncRule from "../../../models/syncRule";
+import i18n from "../../../locales";
+import { Toggle } from "../../toogle/Toogle";
+import { getMetadata } from "../../../utils/synchronization";
+import { getBaseUrl } from "../../../utils/d2";
+import { D2, ModelDefinition } from "../../../types/d2";
+import _ from "lodash";
+import Dropdown from "../../dropdown/Dropdown";
+import { d2ModelFactory } from "../../../models/d2ModelFactory";
+import { D2Model } from "../../../models/d2Model";
+import { MultiSelector } from "d2-ui-components";
+
+interface IncludeExcludeSelectionStepProps {
+    classes: any;
+    d2: D2;
+    syncRule: SyncRule;
+    onChange: (syncRule: SyncRule) => void;
+}
+
+interface ModelSelectItem {
+    name: string;
+    id: string;
+}
+
+interface friendlyRulesNamesDiccionary {
+    [metadataType: string]: string;
+}
+
+const includeExcludeRulesFriendlyNames: friendlyRulesNamesDiccionary = {
```

Comment:
Not sure if it's the time of doing this abstraction but could we build the dictionary with a ``_(includeExcludeRule).split(".").reverse().map(getModelDisplayName).join(" of ").value()``

Reply:
I have doubts if it would be grammatically correct the dictionary result items using this code, anyway, friendly names for rules are pending @adrianq revision, I think after revision it might be time to think if it makes sense to try to obtain the dictionary using an algorithm.